openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py

```diff
@@ -77,7 +79,199 @@ def delete_obsolete_constraints(env):
         openupgrade.delete_records_safely_by_xml_id(
             env, [module + ".constraint_" + table + "_" + name]
         )
-    openupgrade.remove_tables_fks(env.cr, ["mail_moderation"])
+    openupgrade.remove_tables_fks(
+        env.cr,
+        [
+            "mail_moderation",
+            "mail_channel_moderator_rel",
+            "mail_message_mail_channel_rel",
+        ],
+    )
+
+
+def migration_to_mail_group(env):
+    env.cr.execute("SELECT 1 FROM mail_moderation LIMIT 1")
+    has_moderation = env.cr.rowcount
+    env.cr.execute("SELECT 1 FROM mail_channel WHERE email_send LIMIT 1")
+    has_old_mail_groups = env.cr.rowcount
+    env.cr.execute(
+        """
+        SELECT 1 FROM ir_module_module
+        WHERE name = 'portal' AND state in ('installed', 'to upgrade', 'to install')""",
+    )
+    has_portal = env.cr.rowcount
```

Comment:
Why not doing this if no portal?

Reply:
Because mail_group depends on portal module, and if portal is not installed, then we should avoid the installation of portal, right?

Reply:
Ok